experiment/src/org/labkey/experiment/api/property/DomainImpl.java

```diff
@@ -408,6 +409,22 @@ public Lock getDatabaseLock()
         return getLock(_dd);
     }
 
+    @Override
+    public void lockForDelete(DbSchema expSchema)
+    {
+        // NOTE code relies on the lock returned from Domain.getLock() does not require unlock().
+        var lock = getDatabaseLock();
+        assert lock instanceof DbScope.ServerLock;

```

Comment:
Maybe we should also assert that we're inside of a transaction?
```suggestion
        assert lock instanceof DbScope.ServerLock;
        assert ExperimentService.get().getSchema().getScope().isTransactionActive();
```